app/MMB/src/navigation/__tests__/TravelDocumentModalScreen.test.js

```diff
@@ -1,84 +1,94 @@
 });
 
+const getComponent = props =>
+  // $FlowExpectedError: Passing just props needed to run test
+  new TravelDocumentModalScreen({
+    ...props,
+    navigation,
+  });
+
 describe('TravelDocumentModalScreen', () => {
-  it('should validate onIdNumberChange correctly', () => {
-    const instance = getInstance();
-    expect(instance.state.error.idNumber).toBe(false);
+  it('should call navigation setParams when a idNumber changes', () => {
+    const Component = getComponent({
+      idNumber: 'a',
+    });
+
+    // $FlowExpectedError: Passing just props needed to run test
+    Component.componentDidUpdate({ idNumber: 'ab' });
+    expect(navigation.setParams).toBeCalledWith({ disabled: true });
+  });
 
-    instance.onIdNumberChange('1');
-    expect(instance.state.error.idNumber).toBe(true);
+  it('should call navigation setParams when a expiryDate changes', () => {
+    const Component = getComponent({
+      idNumber: 'a',
+    });
 
-    instance.onIdNumberChange('12345');
-    expect(instance.state.error.idNumber).toBe(false);
+    // $FlowExpectedError: Passing just props needed to run test
+    Component.componentDidUpdate({ expiryDate: new Date() });
+    expect(navigation.setParams).toBeCalledWith({ disabled: true });
   });
 
-  it('should validate onDateChange correctly', () => {
-    const instance = getInstance();
-    expect(instance.state.error.expiryDate).toBe(true);
+  it('should call navigation setParams when a noExpiry changes', () => {
+    const Component = getComponent({
+      noExpiry: false,
+      idNumber: '',
+    });
 
-    instance.onDateChange(new Date());
-    expect(instance.state.error.expiryDate).toBe(false);
+    // $FlowExpectedError: Passing just props needed to run test
+    Component.componentDidUpdate({ noExpiry: true, idNumber: '' });
+    expect(navigation.setParams).toBeCalledWith({ disabled: true });
   });
 
-  it('should validate onNoExpiryChange correctly', () => {
-    const instance = getInstance();
-    expect(instance.state.error.expiryDate).toBe(true);
-
-    instance.onDateChange(new Date());
-    expect(instance.state.error.expiryDate).toBe(false);
+  it('should not call set params if idNumber and expiry date remain unchanged', () => {
+    const Component = getComponent({
+      idNumber: 'a',
+      expiryDate: null,
+    });
+
+    // $FlowExpectedError: Passing just props needed to run test
+    Component.componentDidUpdate({
+      idNumber: 'a',
+      expiryDate: null,
+    });
+    expect(navigation.setParams).not.toHaveBeenCalled();
+  });
 
-    instance.onNoExpiryChange(false);
-    expect(instance.state.error.expiryDate).toBe(true);
+  it('should enable save button if idNumber > 5 && expiryDate is set', () => {
+    const Component = getComponent({
+      idNumber: '123456',
+      expiryDate: new Date(),
+    });
+
+    // $FlowExpectedError: Passing just props needed to run test
+    Component.componentDidUpdate({
+      idNumber: '123456',
+      expiryDate: null,
```

Comment:
So you are passing a null expiryDate and you want the Save button not to be disabled? I would have believed it should be disabled in that case, since noExpiry is not set to `true`.

Reply:
I get that this is a bit confusing, but it is actually previousProps that are being passed into componentDidUpdate, so you kind of have to read this backwards.

Reply:
Oh, right. That makes sense.